chart-templates/crd-base/overlay-upstream/templates/validate-install-crd.yaml

```diff
@@ -0,0 +1,12 @@
+# {{- \$found := dict -}}
+${set_found_crd}
+# {{- range .Capabilities.APIVersions -}}
+# {{- if hasKey \$found (toString .) -}}
+# 	{{- set \$found (toString .) true -}}
+# {{- end -}}
+# {{- end -}}
+# {{- range \$_, \$exists := \$found -}}
+# {{- if (eq \$exists false) -}}
+# 	{{- required "Required CRDs are missing. Please install the ${name}-crd chart before installing this chart." "" -}}
+# {{- end -}}
+# {{- end -}}
```

Comment:
I'm not a fan of a few things with this part: 
* it disable's `helm template` without finding the API's, and finding them is not practical, so it slows down development and debugging
* it necessitates a second "overlay" concept, the more concepts we have the harder it is to understand and maintain. 
* Removing this section also removes the concept of setting CRDs to environment variables in the script.
* While the error message is helpful, we should simply be documenting this info in the README/overview pages, and the existing helm error message should be somewhat similar.

Reply:
I agree with all the points you've listed, but I'm not sure if we want to remove this part of the script. Since using the charts directly as Helm charts (rather than installing via Dashboard) is still in scope for users, removing this script would rely on Helm's rendering engine to figure out if CRDs aren't installed on render (with a less direct error message) instead of enforcing that all the CRDs mentioned need to exist on an install. 

However, Helm's rendering engine will only fail if the chart tries to create a custom resource that does not exist on chart deploy; for example, an ECM cluster that has CRDs for Prometheus, Alertmanager, ServiceMonitors, and PrometheusRules will successfully deploy the `rancher-monitoring` chart, but will technically lack full functionality since PodMonitors and ThanosRulers do not exist.

Reply:
To workaround the `helm template` issue I usually just use `helm install <args> --debug --dry-run` or just specify the CRDs like ` --api-versions monitoring.coreos.com/v1/Alertmanager` on `helm template`. One option would be to modify this script to output the necessary helm template command for a dev to use as part of the error message if the CRDs don't exist.

Reply:
We can leave for now and see if it turns into a headache. I think we should re-asses on rewrite.